internal/internal.go

```diff
@@ -238,6 +240,11 @@ var (
 	// SetBufferPoolingThresholdForTesting updates the buffer pooling threshold, for
 	// testing purposes.
 	SetBufferPoolingThresholdForTesting any // func(int)
+
+	// HTTPSProxyFromEnvironmentForTesting returns the URL of the proxy to use
+	// for testing purposes. It is used to override the `http.ProxyFromEnvironment`
+	// function for testing purposes.
+	HTTPSProxyFromEnvironmentForTesting func(*http.Request) (*url.URL, error)
```

Comment:
You can move these to a new `resolver/delegatingresolver/internal` module since only delegating resolver tests need to use it. See the [pickfirst/internal](https://github.com/grpc/grpc-go/blob/master/balancer/pickfirst/internal/internal.go) package as an example.

Reply:
As discussed offline, revert this back to `any` and remove the `"net/http", "net/url"` imports and the dependency tests should pass again.

Reply:
This is change is from the other PR. Resolving.